sample/python-timeout/job.py

```diff
@@ -36,6 +38,14 @@ def perform(self, mode: str = "timeout", t: int = 250) -> dict:
         elif mode == "crash":
             raise Exception("Windows12 has stopped working")
 
+        elif mode == "exit":
+            sys.exit()
+
+        elif mode == "oom":
+            for x in range(10 * 1024 * 1024):
+                self.mem.append(x)
+            return {'mem': len(self.mem)}
```

Comment:
This is not a foolproof way to go oom. I don't know how much memory it does take up, but personally, I'd suggest `while True` rather than a `for` loop. That way it's guaranteed that it'll go oom no matter what.

There's of course a chance that what is written takes such a ridiculous amount of memory that this is merely a technicality. If so, feel free to ignore the comment.

Reply:
This name `oom` might be misleading. This thing is actually trying to handle the different use case. I don't want it to always go OOM-killed, but for the second time it's called.

I wanted to test retrying a Job call when a job gets OOM-killed, it's hard to test it if a Job is always killed even after the restart. That's why I've made this sample job to allocate more and more memory with every request (but not too much, in order not to kill it with the first request). So once it's killed, it survives the retried attempt.

Reply:
I would rename it or chuck in a comment explaining that then.

Reply:
Alright, I'm gonna rename it to `malloc`